base_layer/core/src/transactions/transaction_components/encrypted_openings_s.rs

```diff
@@ -0,0 +1,211 @@
+    pub fn from_bytes(bytes: &[u8]) -> Result<Self, EncryptedOpeningsError> {
+        if bytes.len() != SIZE {
+            return Err(EncryptedOpeningsError::IncorrectLength(format!(
+                "Expected {} bytes, got {}",
+                SIZE,
+                bytes.len()
+            )));
+        }
+        let mut data_1: [u8; BORSH_32] = [0u8; BORSH_32];
+        data_1.copy_from_slice(bytes.get(..BORSH_32).ok_or_else(|| EncryptedOpeningsError::IncorrectLength(
+            "Out of bounds 'data_1'".to_string(),
+        ))?);
+        let mut data_2: [u8; BORSH_X] = [0u8; BORSH_X];
+        data_2.copy_from_slice(
+            bytes
+                .get(BORSH_32..SIZE)
+                .ok_or_else(|| EncryptedOpeningsError::IncorrectLength(
+                    "Out of bounds 'data_2'".to_string(),
+                ))?,
+        );
+        Ok(Self { data_1, data_2 })
+    }
+
+    /// Custom convert `EncryptedOpenings` to byte vector
+    pub fn as_byte_vector(&self) -> Vec<u8> {
+        let mut bytes = Vec::with_capacity(SIZE);
+        bytes.extend_from_slice(&self.data_1);
+        bytes.extend_from_slice(&self.data_2);
+        bytes
+    }
+}
+
+impl Default for EncryptedOpeningsS {
+    fn default() -> Self {
+        Self {
+            data_1: [0u8; BORSH_32],
+            data_2: [0u8; BORSH_X],
+        }
+    }
+}
+
+/// EncryptedOpenings errors
+#[derive(Debug, Error)]
+pub enum EncryptedOpeningsError {
+    #[error("Encryption failed: {0}")]
+    EncryptionFailed(Error),
+    #[error("Conversion failed: {0}")]
+    ByteArrayError(#[from] ByteArrayError),
+    #[error("Incorrect length: {0}")]
+    IncorrectLength(String),
+}
+
+// Chacha error is not StdError compatible
+impl From<Error> for EncryptedOpeningsError {
+    fn from(err: Error) -> Self {
+        Self::EncryptionFailed(err)
+    }
+}
+
+impl EncryptedOpeningsS {
+    const TAG: &'static [u8] = b"TARI_AAD_VALUE_AND_MASK";
+
+    /// Encrypt the value and mask (with fixed length) using ChaCha20-Poly1305 with a fixed zero nonce to save space -
+    /// this design assumes a unique `encryption_key` every time and therefore uses a fixed nonce
+    pub fn encrypt_openings(
+        encryption_key: &PrivateKey,
+        commitment: &Commitment,
+        value: MicroTari,
+        mask: &PrivateKey,
+    ) -> Result<EncryptedOpeningsS, EncryptedOpeningsError> {
+        let mut openings = value.as_u64().to_le_bytes().to_vec();
+        openings.append(&mut mask.to_vec());
+        let aead_payload = Payload {
+            msg: openings.as_slice(),
+            aad: Self::TAG,
+        };
+        let aead_key = kdf_aead(encryption_key, commitment);
+        let cipher = ChaCha20Poly1305::new(GenericArray::from_slice(aead_key.reveal()));
+        let ciphertext = cipher.encrypt(&Nonce::default(), aead_payload)?;
+
+        EncryptedOpeningsX::from_bytes(ciphertext.as_slice())
+    }
+
+    /// Authenticate and decrypt the value and mask
```

Comment:
Recommend a comment that this design is not key committing, so the caller should not rely solely on successful decryption to assert the expected key was used.